src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -131,7 +131,20 @@ protected ActionResult execute() throws EntityDoesNotExistException {
                     continue;
                 }
 
-                responsesRecipients.add(response.recipient);
+                // If there are duplicate values in recipient list (except empty strings),
+                // trigger this error.
+                // Note: If response for a recipient is left out, the recipient is then empty string,
+                // due to partial submission feature, there can be multiple recipient with no resposnes,
+                // in that case, there can be multiple empty strings which should not be counted.
+                if ("".equals(response.recipient)) {
+                    responsesRecipients.add(response.recipient);
+                } else {
+                    if (responsesRecipients.contains(response.recipient)) {
+                        errors.add(String.format(Const.StatusMessages.FEEDBACK_RESPONSE_DUPLICATE_RECIPIENT, questionIndx));
+                    }
+                    responsesRecipients.add(response.recipient);
```

Comment:
So the duplicate recipient would still be added to `responseRecipients`? Does it make a difference if we just `continue` if there's an error (like in line `131` above)?

Reply:
It's OK I think. After all, a legit user won't see the error.